### PR TITLE
mrpeek: move focus with left mouse click & drag

### PR DESCRIPTION
**Pull request by jdtournier**
Not ideal, but probably the best we can offer given the limitations of
mouse handling on a VT...

### CODE REVIEW

File: cmd/mrpeek.cpp [hunk 1]
```diff
@@ -288,6 +288,7 @@ void run ()
         case '+': image_scale *= 1.1; std::cout << VT::ClearScreen; break;
         case '-': image_scale /= 1.1; std::cout << VT::ClearScreen; break;
         case VT::MouseMoveLeft: focus[x_axis] += xp-x; focus[y_axis] += yp-y; break;
+        case VT::Escape: colourmap.update_scaling (x, y); break;
```
Comment (jdtournier) [inline cmd/mrpeek.cpp]: This doesn't seem to work... But more to the point, the Home key already does this. See commit 5271a76.

Comment (jdtournier) [reply cmd/mrpeek.cpp]: Actually, the Esc key seems to be handled reliably all of a sudden... I'll fix it to use the Esc key and remove the Home key binding.

**Comment by jdtournier** [comment]
The changes in a8971ae should really help for laggy systems. Would appreciate feedback on that one!

**Comment by maxpietsch** [comment]
> The changes in [a8971ae](https://github.com/MRtrix3/mrpeek/commit/a8971aed1d4d254dca9d01f6d021e04add6babc0) should really help for laggy systems. Would appreciate feedback on that one!

I think it is faster in iTerm2 but not blistering. mlterm seems similar.

File: cmd/mrpeek.cpp (hunk 1 above)
Comment (maxpietsch) [reply cmd/mrpeek.cpp]: I added it as Home is Fn + Left Arrow on a macbook and the right click mouse drag was quite slow still

**Comment by jdtournier** [comment]
>> The changes in a8971ae should really help for laggy systems. Would appreciate feedback on that one!
>
> I think it is faster in iTerm2 but not blistering. mlterm seems similar.

I don't expect it to be faster, but that event processing stops more or less as soon as you've stopped pressing the key or moving the mouse. It's purely about avoiding long waits while your display catches up with the backlog of events...

**Comment by jdtournier** [comment]
OK, let's merge this to master, and discuss the next phase of the project.

**Comment by dchristiaens** [comment]
This is starting to look very nice. Moving the focus now works on iTerm2, though the lag is annoying. However, intensity scaling with the mouse action no longer works.

**Comment by maxpietsch** [comment]
before merging to master, do you want to merge #11 first into this branch?

> However, intensity scaling with the mouse action no longer works.

yes, same here. It did work before.

**Comment by jdtournier** [comment]
> However, intensity scaling with the mouse action no longer works

That's odd... Still works fine here. Not sure what's going on...

**Comment by maxpietsch** [comment]
the right-click event doesn't get to`switch (event)`, left-click does. instead, I get an image save dialogue

**Comment by jdtournier** [comment]
So I'm guessing the event is captured by your terminal, and never gets forwarded to the application?

**Comment by jdtournier** [comment]
How did it ever work before...?!?

**Comment by maxpietsch** [comment]
it rarely works. I went all the way back to 1bfc5cce3bf4447b020fce85e2c73aefc210cffa and the behaviour seems to be the same. restarted iTerm does not help. mlterm is fine.

**Comment by jdtournier** [comment]
Are there any options to iTerm that you can tweak?

**Comment by jdtournier** [comment]
This suggests there's options to configure what the right click does:
https://dbavic.wordpress.com/tag/right-click-paste-on-iterm2/

**Comment by maxpietsch** [comment]
bingo right click is captured by iTerm2 (most of the time), right-click and drag works when the key binding is changed:
image

**Comment by maxpietsch** [comment]
shall we add `shift` + `MouseMoveLeft` as an alternative to `MouseMoveRight` or are key and mouse events not supported?

**Review by maxpietsch — APPROVED** [review]

**Comment by jdtournier** [comment]
Supported, but captured by most terminals - at least on mine, for one...

**Comment by jdtournier** [comment]
I can get Alt+mouse events though...

**Comment by jdtournier** [comment]
I was thinking we might need to figure out some key bindings for brightness/contrast and focus handling, just to make sure it's usable without a mouse.

**Comment by dchristiaens** [comment]
> I was thinking we might need to figure out some key bindings for brightness/contrast and focus handling, just to make sure it's usable without a mouse.

That would be great!

Also, if a single left mouse click can not be captured to move the focus, could something like ctrl-click work instead?

**Comment by maxpietsch** [comment]
alt+mouse might work
focus: "x" or "y" + voxel number + "enter"

ctrl click does open the menu in iTerm2
not sure what the default is:
image

**Review by dchristiaens — APPROVED** [review]

**Pull request merged**